targets/default/jerry-port-default-jobqueue.c

```diff
@@ -0,0 +1,139 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerry-port.h"
+#include "jerry-port-default.h"
+#include "jrt.h"
+#include "jmem.h"
+#include "jerryscript.h"
+
+#ifndef CONFIG_DISABLE_ES2015_PROMISE_BUILTIN
+
+typedef struct jerry_port_queueitem_t jerry_port_queueitem_t;
+
+/**
+ * Description of the queue item.
+ */
+struct jerry_port_queueitem_t
+{
+  jerry_port_queueitem_t *next_p; /**< point to next item */
+  jerry_job_handler_t handler; /**< the hanlder for the job*/
+  void *job_p; /**< point to the job */
+};
+
+/**
+ * Description of a job queue (FIFO)
+ */
+typedef struct
+{
+  jerry_port_queueitem_t *head_p; /**< point to the head item of the queue */
+  jerry_port_queueitem_t *tail_p; /**< point to the tail item of the queue*/
```

Comment:
`/**< points to the head item of the queue */`
`/**< points to the tail item of the queue */`